tests/foreman/api/test_parameters.py

```diff
@@ -14,44 +14,84 @@
 import pytest
 
 
+@pytest.fixture
+def param_name():
+    """Generate name string for common parameter and return to test cases"""
+    """Generate name for common parameter"""
+    return gen_string('alpha')
+
+
+@pytest.fixture
+def param_value():
+    """Generate value string for common parameter and return to test cases"""
+    """Generate value for common parameter"""
+    return gen_string('alpha')
+
+
+@pytest.fixture
+def cp(param_name, param_value, module_target_sat):
+    """Create common parameter object, yields it for test cases, delete object before teardown"""
+    """Create common parameter"""
+    cp = module_target_sat.api.CommonParameter(name=param_name, value=param_value).create()
+
+    yield cp
+
+    cp.delete()
+
+
+@pytest.fixture
+def host(param_name, param_value, module_org, module_location, module_target_sat, cp, hostgroup):
```

Comment:
Many parameters here appear to be unused.
```suggestion
def host(module_org, module_location, module_target_sat):
```